sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -664,19 +669,61 @@ func tagImage(ctx context.Context, imageName string, newImageName string, client
 
 // withDynamicResolver provides an initialized resolver for use with ref.
 func withDynamicResolver(ctx context.Context, ref string) containerd.RemoteOpt {
-	if !strings.HasPrefix(ref, "ecr.aws/") {
-		// not handled here
-		return func(_ *containerd.Client, _ *containerd.RemoteContext) error { return nil }
-	}
+	if strings.HasPrefix(ref, "ecr.aws/") {
```

Comment:
nit: I have a slight bias toward using `switch` statements instead of if/else chains

```go
switch {
case strings.HasPrefix(ref, "ecr.aws/"):
	// body
case strings.HasPrefix(ref, "public.ecr.aws/"):
	// body
default:
	// body
}
```